src/index.ts

```
@@ -1,34 +1,36 @@
-class User {
+interface User {
   name: string
   favoriteColor: string
   age: number
-  id: number
-
-  constructor(name: string, favoriteColor: string, age: number, id: number) {
-    this.name = name
-    this.favoriteColor = favoriteColor
-    this.age = age
-    this.id = id
-  }
+  id?: number
 }
 
 class UserAPI {
   users: Array<User>
+  currentId: number
 
   constructor() {
     this.users = []
+    this.currentId = 0
   }
 
   private assignId() {
-
+    this.currentId++
+    return this.currentId - 1
   }
 
   addUser(user: User) {
-
+    user.id = this.assignId()
+    this.users.push(user)
   }
 
   getUserById(id: number) {
```

rather than return a string or a user object, throw an error if the user is not found (see javascript.info on how to throw errors)

---

make sure to intentionally induce the error in the console log tests below